doc/user_guide/io.rst

```diff
@@ -1030,6 +1032,162 @@ HDF5 file. All other keyword arguments will be passed to
+can contain a vary large number of datasets.
+
+Extra loading arguments
++++++++++++++++++++++++
+- ``dset_search_keys`` : ``str`` or ``list`` 
+Absolute path(s) or substring(s) to use to find
+one or more datasets. (default is ``None`` all Nexus Datasets will be read)
+- ``meta_search_keys`` : ``str`` or ``list`` of strings 
+Absolute path(s) or substring(s) to use to find
+one or more datasets. (default is ``None`` all Nexus Metadata will be read)
+ 
+Reading the sole Nexus dataset within a Nexus file:
+
+.. code-block:: python
+
+    >>> hs.load("sample.nxs")
+
+If multiple datasets are present within the Nexus file and you try the same 
+command again **all** available NXdata datasets will be loaded.
+To load NXdata and hdf datasets as well
+
+.. code-block:: python
+
+    >>> hs.load("sample.nxs",nxdata_only=False)
+
+.. note::
+
+    Given that HDF5 files can accommodate very large datasets, setting 
+    ``lazy=True`` is strongly recommended if the contents of the HDF5 file 
+    are not known apriori. This prevents issues with regard to loading datasets
+    far larger than memory.
+
+    Also note that setting ``lazy=True`` leaves the file handle to the 
+    HDF5 file open.
+    If it is important that the files be closed after reading,
+    set ``lazy=False``.
+
+.. code-block:: python
+
+    >>> hs.load("sample.h5")
+    [<Signal2D, title: HAADF, dimensions: (|128, 128)>,
+    <Signal1D, title: EELS, dimensions: (|64, 64, 1024)>]
+
+We can load a specific datasets using the ``dset_search_path`` keyword argument. 
+Setting it to the absolute path of the desired dataset will cause 
+the single dataset to be loaded.
+Setting it to a substring or list of substring, datasets who's paths contain 
+one of the substrings will be loaded.
+
+
+.. code-block:: python
+
+    >>> # Loading a specific dataset
+    >>> hs.load("sample.h5", dset_search_path='/entry/experiment/EDS/data')
+
+We can also choose to load only specific items from the metadata using
+ ``dset_path`` keyword argument. This can be a list of 
+absolute paths of the desired dataset or substrings to find in the metadata
+path.
+
+.. code-block:: python
+
+    >>> # Loading a specific dataset
+    >>> hs.load("sample.h5", dset_search_path='EDS')
+
+Nexus files also support parameters or dimensions that have been varied 
+non-linearly.
+Since HyperSpy Signals expect linear variation of parameters / axes, such 
+non-linear information would be lost in the axes manager and replaced with
+indices
+
+
+Writing
+^^^^^^^
+Signals can be written to new Nexus files using the standard ``.save()`` 
+function.
+When Hyperspy creates a signal it creates and holds Hyperspy metadata 
+and original_metadata seperately.
+When we load a Nexus file the Nexus metadata is stored in 
+original_metadata. Conversely when we save to Nexus only the 
+original_metadata is currently stored to the Nexus file along with 
+the signal data and axes. 
```

Comment:
So s.save("output.nxs") loses any hyperspy metadata in this implementation? Is it a total nightmare to store that metadata in nexus? If it can be saved I think it would be preferable, but I appreciate that might not be viable right now.

Reply:
Fixed - to save hyperspy_metadata and restore it from nexus